common4j/src/main/com/microsoft/identity/common/java/telemetry/AbstractTelemetryContext.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.telemetry;
+
+import com.microsoft.identity.common.java.interfaces.IKeyPairStorage;
+import lombok.NonNull;
+
+import java.util.TimeZone;
+
+import static com.microsoft.identity.common.java.telemetry.TelemetryEventStrings.App;
+import static com.microsoft.identity.common.java.telemetry.TelemetryEventStrings.Device;
+import static com.microsoft.identity.common.java.telemetry.TelemetryEventStrings.Os;
+
+/**
+ * TelemetryContext is a dictionary of information about the state of the device.
+ * It is attached to every outgoing telemetry calls.
+ */
+public abstract class AbstractTelemetryContext extends Properties {
+    private TelemetryPropertiesCache mTelemetryPropsCache;
+
+    protected AbstractTelemetryContext(@NonNull final IKeyPairStorage telemetryPropertyCacheStorage) {
+        super();
+        mTelemetryPropsCache = new TelemetryPropertiesCache(telemetryPropertyCacheStorage);
```

Comment:
Rather than have this constructor call another constructor (painful to mock/test) can we pass-in the `TelemetryPropertiesCache` instance directly? This makes the "has-a" relationship clearer and makes mocking easier